### PR TITLE
Feat/active chats panel

### PR DESCRIPTION
**Pull request by crui3er**
- when active chat list exceeds limit (max 4 items) on adding a new item, then less important chat is eliminated from the list;
- stopping recording/listening on a chat, does not remove the chat from the active chat list immediately;
image

### CODE REVIEW

File: src/dotnet/Chat.UI.Blazor/Services/ChatUI.cs [hunk 1]
```diff
@@ -198,4 +262,100 @@ private Task<ISyncedState<long>> CreateLastReadEntryState(Symbol chatId, Cancell
...
+                limit--; // reserve one slot for recording chat
+        }
+
+        if (listeningChatIds.Count < limit)
+            return Symbol.Empty;
+
+        bool IsPinnedChat(Symbol c)
+            => PinnedChatIds.Value.Keys.Contains(c.Value, StringComparer.OrdinalIgnoreCase);
+
+        async Task<Symbol> GetChatToEliminate(ICollection<Symbol> chatIds)
```
Comment (alexis-kochetov) [inline src/dotnet/Chat.UI.Blazor/Services/ChatUI.cs:281]: Let's put inner methods at the end of the parent method

File: src/dotnet/Chat.UI.Blazor/Services/ChatUI.cs [hunk 2]
```diff
@@ -42,17 +53,20 @@ public ChatUI(IServiceProvider services)
...
             });
-        RecordingChatId = StateFactory.NewKvasStored<Symbol>(new(localSettings, nameof(RecordingChatId)));
-        ListeningChatIds = StateFactory.NewKvasStored<ImmutableList<Symbol>>(
+        _recordingChatId = StateFactory.NewKvasStored<Symbol>(new(localSettings, nameof(RecordingChatId)));
+        _listeningChatIds = StateFactory.NewKvasStored<ImmutableList<Symbol>>(
             new(localSettings, nameof(ListeningChatIds)) {
                 InitialValue = ImmutableList<Symbol>.Empty,
                 Corrector = FixListeningChatIds,
             });
+        _activeChatIds = StateFactory.NewMutable(ImmutableList<(Symbol, Moment)>.Empty);
```
Comment (alexis-kochetov) [inline src/dotnet/Chat.UI.Blazor/Services/ChatUI.cs:67]: I think we should use some synced state there and store active chats at the server - we would like to have the synced across all devices

**Review by alexis-kochetov — CHANGES_REQUESTED** [review]

File: src/dotnet/Chat.UI.Blazor/Services/ChatUI.cs (hunk 1 above)
Comment (crui3er) [reply src/dotnet/Chat.UI.Blazor/Services/ChatUI.cs:344]: done

File: src/dotnet/Chat.UI.Blazor/Services/ChatUI.cs (hunk 2 above)
Comment (crui3er) [reply src/dotnet/Chat.UI.Blazor/Services/ChatUI.cs]: done

**Review by alexis-kochetov — APPROVED** [review]

**Review by alexis-kochetov — APPROVED** [review]

**Pull request merged**